### PR TITLE
[Exp] Add Log Expression

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
This PR removes `UnaryAbsExpression` and creates a more generic `UnaryExpression` which is able to get `Abs` and `Log`, and can be easily extended in future for other operators if required.

Finally I am also adding documentation in the same PR :)

**🆕 Changelog**
- Removes `UnaryAbsExpression`
- Adds `UnaryExpression`
- Adds `ExpressionUtils::Log`
- Adds tests
- Adds **documentation**

### CODE REVIEW

File: kratos/expression/unary_expression.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        static_assert(!std::is_same_v<TOperationType, TOperationType>, "Unsupported unary operation type.");
+    }
+
+    msg << "(" << mpSourceExpression << ")";
+    return msg.str();
+}
+
+// template instantiations
+template class UnaryExpression<UnaryOperations::Absolute>;
+template class UnaryExpression<UnaryOperations::Logarithmic>;
```
Comment (matekelemen) [inline kratos/expression/unary_expression.cpp]:
```suggestion
template class UnaryExpression<UnaryOperations::Logarithm>;
```

Comment (sunethwarna) [reply kratos/expression/unary_expression.cpp]: done :)

File: kratos/expression/unary_expression.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+template <class TOperationType>
+std::string UnaryExpression<TOperationType>::Info() const
+{
+    std::stringstream msg;
+
+    if constexpr(std::is_same_v<TOperationType, UnaryOperations::Absolute>) {
+        msg << "Abs";
+    } else if constexpr(std::is_same_v<TOperationType, UnaryOperations::Logarithmic>) {
```
Comment (matekelemen) [inline kratos/expression/unary_expression.cpp]:
```suggestion
    } else if constexpr(std::is_same_v<TOperationType, UnaryOperations::Logarithm>) {
```

Comment (sunethwarna) [reply kratos/expression/unary_expression.cpp]: missed it :D

**Comment by sunethwarna** [comment]
ping @matekelemen

**Review by matekelemen — APPROVED** [review]
thx

**Pull request merged**